### DISCUSSION TITLE
Determining the System Upgrade Controller Version Used by Rancher for RKE2 Upgrades

### DISCUSSION THREAD

**Discussion by LucEast** [rancher/system-upgrade-controller#338] [General]
Hello everyone,

I’m currently running Rancher v2.7.4 to manage an RKE2 cluster on Kubernetes v1.24.10. I’m planning to upgrade this cluster to Kubernetes v1.25.16 using Rancher’s Cluster Management GUI.

Since Rancher utilizes the System Upgrade Controller to handle upgrades, I want to understand how to determine which version of the System Upgrade Controller Rancher will use during this process.

Some specific questions I have:

- Does Rancher automatically manage the System Upgrade Controller version based on the target Kubernetes version?
- Is there a way to find out which version of the System Upgrade Controller Rancher will deploy for a specific upgrade path (e.g., v1.24.10 to v1.25.16)?

Any insights, documentation references, or best practices for managing this would be greatly appreciated!

Thanks in advance!

**Comment by brandond** [comment]
The answer is "its complicated", but yes Rancher manages the version of SUC deployed to the imported cluster. However, it only deploys or upgrades the SUC deployment (if necessary) when the downstream cluster version is changed, as a prerequisite for upgrading the cluster. It does not keep it updated between cluster version changes.

**Reply by brandond** [reply]
The deployed version is not just based on the rancher version, its based on the version of Kubernetes that the downstream cluster is currently running. If you look on the Rancher cluster you can check:

```
> kubectl get catalogtemplateversions -n cattle-global-data -l catalog.cattle.io/template_name=system-library-rancher-k3s-upgrader -o custom-columns=NAME:metadata.name,VERSION:.spec.version,KUBEVERSION:.spec.kubeVersion
NAME                                        VERSION   KUBEVERSION
system-library-rancher-k3s-upgrader-0.1.0   0.1.0     <none>
system-library-rancher-k3s-upgrader-0.2.0   0.2.0     <none>
system-library-rancher-k3s-upgrader-0.2.1   0.2.1     <none>
system-library-rancher-k3s-upgrader-0.3.0   0.3.0     <none>
system-library-rancher-k3s-upgrader-0.3.1   0.3.1     >= 1.16.0-0
system-library-rancher-k3s-upgrader-0.3.2   0.3.2     >= 1.16.0-0
system-library-rancher-k3s-upgrader-0.3.3   0.3.3     >= 1.16.0-0
system-library-rancher-k3s-upgrader-0.4.0   0.4.0     >= 1.16.0-0
system-library-rancher-k3s-upgrader-0.5.0   0.5.0     >= 1.23.0-0
system-library-rancher-k3s-upgrader-0.6.0   0.6.0     >= 1.23.0-0
system-library-rancher-k3s-upgrader-0.7.0   0.7.0     >= 1.25.0-0
```

That’s the version of the rancher-k3s-upgrader chart that will get deployed. https://github.com/rancher/system-charts/tree/dev-v2.9/charts/rancher-k3s-upgrader is where you could cross-walk that to an actual SUC version:

0.3.3=v0.9.1
0.4.0=v0.10.0
0.6.0=v0.13.1
0.7.0=v0.13.4

**Comment by dweomer** [comment]
> I’m currently running Rancher v2.7.4 to manage an RKE2 cluster on Kubernetes v1.24.10. I’m planning to upgrade this cluster to Kubernetes v1.25.16 using Rancher’s Cluster Management GUI.

Then do that, manage your upgrade via the Rancher UI/CRD. Forget about SUC, it is an implementation detail that is not under your control.
